lib/rule-tester/rule-tester.js

```diff
@@ -991,6 +992,7 @@ class RuleTester {
                             assert.strictEqual(message.endColumn, error.endColumn, `Error endColumn should be ${error.endColumn}`);
                         }
 
+                        assert.ok(!message.suggestions || hasOwnProperty(error, "suggestions"), `Error should have suggestions on error with message: "${message.message}`);
```

Comment:
We should also check this for error tests that specify only string/regex.

For example, if we have a test case like this for the `no-array-constructor` rule, it should fail because the error has suggestions:

```js
{
    code: "new Array()",
    errors: ["The array literal notation [] is preferable."]
},
```